styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -142,6 +144,15 @@ public void receive(Event event) throws IsClosed {
   }
 
   @Override
+  public Map<WorkflowInstance, RunState> backfillActiveStates() {
+    final ImmutableMap.Builder<WorkflowInstance, RunState> builder = ImmutableMap.builder();
+    states.entrySet().stream()
+        .filter(entry -> isBackfill(entry.getValue().runState.data().trigger().get().toTrigger()))
```

Comment:
Potential exception thrown here because of the unchecked `.get()` on the optional.